diff --git a/tfa.module b/tfa.module
index 043df03..b052904 100644
--- a/tfa.module
+++ b/tfa.module
@@ -44,6 +44,11 @@ function tfa_entry_access($account, $url_hash) {
  */
 function tfa_permission() {
   return array(
+		'skip tfa' => array(
+			'title' => t('Skip TFA process'),
+			'description' => t('Skip the Two-factor authentication process and authenticate as normal.'),
+			'restrict access' => TRUE,
+    ),
     'require tfa' => array(
       'title' => t('Require TFA process'),
       'description' => t('Require two-factor to authenticate. Note: users without TFA setup will be unable to log in.'),
@@ -201,7 +206,7 @@ function tfa_login_submit($form, &$form_state) {
   $account = isset($form_state['uid']) ? user_load($form_state['uid']) : user_load_by_name($form_state['values']['name']);
   $tfa = tfa_get_process($account);
 
-  if (user_access('require tfa', $account) && !tfa_login_complete($account) && !$tfa->ready()) {
+  if (user_access('require tfa', $account) && !tfa_login_complete($account) && !$tfa->ready() && !tfa_login_allowed($account)) {
     drupal_set_message(t('Login disallowed. You are required to setup two-factor authentication. Please contact a site administrator.'), 'error');
     $form_state['redirect'] = 'user';
   }
@@ -260,6 +265,9 @@ function tfa_login_complete($account) {
  * @return bool
  */
 function tfa_login_allowed($account) {
+	if (user_access('skip tfa', $account)) {
+		return TRUE;
+	}
   // Check if login plugins will allow login.
   $tfa = tfa_get_process($account);
   return $tfa->loginAllowed() === TRUE;
@@ -274,7 +282,7 @@ function tfa_user_login(&$edit, $account) {
   }
 
   $tfa = tfa_get_process($account);
-  if (user_access('require tfa', $account) && !tfa_login_complete($account) && !$tfa->ready()) {
+  if (user_access('require tfa', $account) && !tfa_login_complete($account) && !$tfa->ready() && !tfa_login_allowed($account)) {
     tfa_logout();
     drupal_set_message(t('Login disallowed. You are required to setup two-factor authentication. Please contact a site administrator.'), 'error');
     drupal_goto('user');
